llvm/lib/Target/AIE/AIE2TargetMachine.cpp

```diff
@@ -139,6 +139,10 @@ void AIE2PassConfig::addIRPasses() {
           }));
     }
   }
+
+  if (TM->getOptLevel() > CodeGenOptLevel::None)
+    addPass(createInferAddressSpacesPass());
```

Comment:
Do you know if running the pass helped?

Reply:
Yes, it helps have added a test that shows how it cleans up the IR at early stage `llvm/test/CodeGen/AIE/aie2/GlobalISel/infer-address-space.ll`.

Without this pass we are still meeting the current best for the inner most loop but it helps in the code around.

Reply:
Sounds good :)